gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -575,7 +529,69 @@ private void addConstructsFinalStateToTaskState(InstrumentedExtractorBase<?, ?>
    * 3. Check whether to publish data in task.
    */
   public void commit() {
-    // TODO- refactor the current Task.run() method to run & commit phases, and add commit content here.
+    try {
+      // Check if all forks succeeded
+      boolean allForksSucceeded = true;
+      for (Optional<Fork> fork : this.forks.keySet()) {
+        if (fork.isPresent()) {
+          if (fork.get().isSucceeded()) {
+            if (!fork.get().commit()) {
+              allForksSucceeded = false;
+            }
+          } else {
+            allForksSucceeded = false;
+          }
+        }
+      }
+
+      if (allForksSucceeded) {
+        // Set the task state to SUCCESSFUL. The state is not set to COMMITTED
+        // as the data publisher will do that upon successful data publishing.
+        this.taskState.setWorkingState(WorkUnitState.WorkingState.SUCCESSFUL);
+      } else {
+        LOG.error(String.format("Not all forks of task %s succeeded", this.taskId));
+        this.taskState.setWorkingState(WorkUnitState.WorkingState.FAILED);
```

Comment:
please use failTask() with a exception specifying which forks failed. This will ensure consistent behavior about what failure state is maintained in the task.